Bridge C++ library core to PySCF

pi246 opened the pull request:
Please let me know if there is any modification I should make! Thanks!

pi246 committed:
Bridge C++ library core to PySCF

zc62 commented on src/pyscf_losc/utils.py:
```diff
@@ -0,0 +1,427 @@
+
+    return loscmf
+
+
+
+
```
Extra empty lines

zc62 commented on src/pyscf_losc/losc_options.py:
```diff
@@ -0,0 +1,157 @@
+            for key, val in opts.items():
+                rst.append(f'    {key}: {val}')
+            rst.append('')
+        return '\n'.join(rst)
```
Configure your editor to always insert a newline, even if it is the end of file. This happens to all new files you created in this commit.
If using vscode, configure in this way:
```json
"files.insertFinalNewline": true,
```
Some others I use along with this setting:
```json
"files.trimTrailingWhitespace": true,
"files.trimFinalNewlines": true,
```

Or did you use any code formatter? If so, find corresponding configurations for that.

zc62 commented on src/pyscf_losc/utils.py:
```diff
@@ -0,0 +1,427 @@
+
+    return nocc, occ_idx, occ_val
+
+
+
+
```
Extra empty lines. I think code formatters should catch these?

zc62 commented:
The code looks great, though I cannot really tell if it works as intended or not. I think it would be best if you can add unit tests. @Miocbb How did you do unit tests for the psi4 plugin?

In addition, I think proper documentation about how to use this plugin is required, otherwise people who are not familiar with this cannot easily use it. Refer to how @Miocbb wrote documentations for the psi4 plugin. See a bunch of rst files under https://github.com/Yang-Laboratory/losc/tree/2405d7b1e7d1d8f593403573f6af7a0b6b988c5e/docs_source/sphinx

zc62 commented:
And you can try to write more standardized git commit messages as well. [Here is what I saw and took as a good standard for commit messages.](https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md#write-good-commit-messages) You can also refer to the commit history of this repository and see how Miocbb wrote them.

Miocbb commented:
> The code looks great, though I cannot really tell if it works as intended or not. I think it would be best if you can add unit tests. @Miocbb How did you do unit tests for the psi4 plugin?
> 
> In addition, I think proper documentation about how to use this plugin is required, otherwise people who are not familiar with this cannot easily use it. Refer to how @Miocbb wrote documentations for the psi4 plugin. See a bunch of rst files under https://github.com/Yang-Laboratory/losc/tree/2405d7b1e7d1d8f593403573f6af7a0b6b988c5e/docs_source/sphinx

I made some simple tests for psi4_losc under losc/test/psi4_losc. @pi246

pi246 committed:
src/pyscf_losc: reformat all the files

pi246 committed:
src/pyscf_losc: fixed bug on setting losc options

pi246 committed:
psi4_losc.py: fix bug related to curvature version

CurvatureV1 could not be used for SC calculations
due to a typo on line 250 of psi4_losc.py.
The original line judges if the value of
'curvature' is i, while it should judge
if the value of 'curvature_version' is 1.

zc62 commented:
Not sure why this PR got merged without further review, as apparently 65099b36f97a9dfb7f7aa9b33ecf2407850ca9a5 got back those format issues that e42faebb5608290262f9e375ed1e7cb842cadfaf tried to resolve, for example. @pi246
